src/halmos/__main__.py

```diff
@@ -1002,17 +1002,11 @@ def is_valid_model(model) -> bool:
 def str_model(model, args: argparse.Namespace) -> str:
     def select(var):
         name = str(var)
-        if name.startswith("p_"):
-            return True
-        if name.startswith("halmos_"):
-            return True
-        return False
+        return name.startswith("p_") or name.startswith("halmos_")
 
     select_model = filter(select, model) if not args.print_full_model else model
-    result = ",".join(
-        sorted(map(lambda decl: f"\n    {decl} = {hexify(model[decl])}", select_model))
-    )
-    return f"[{result}]"
+    formatted = [f"\n    {decl} = {hexify(model[decl])}" for decl in select_model]
+    return "".join(sorted(formatted))
```

Comment:
can we return some meaningful value (or message) when the return value is empty string, so that it can be printed out like this?
```
Counterexample: <some-value-or-message>
```

i believe it is useful info, telling users that no specific inputs are involved with this assertion violation.

Reply:
for example, how about returning `"[]"` (without the leading newline) when the filtered list is empty?

Reply:
how about something cute like this? :)

```
[FAIL] check_invariant() (paths: 0/2, time: 0.43s, bounds: [])
Counterexample: ∅
```

Reply:
amazing!

Reply:
ah, there are some issues with handling the unicode empty set symbol `\u2205` in windows:
```
______________________________ test_run_bytecode ______________________________

args = Namespace(root='D:\\a\\halmos\\halmos', contract=None, function='check', loop=2, width=None, depth=None, array_lengths...lse, bytecode=None, reset_bytecode=None, test_parallel=False, symbolic_jump=True, print_potential_counterexample=False)

    def test_run_bytecode(args):
        # sets the flag in the global args for the main module
        args.symbolic_jump = True
    
        hexcode = "34381856FDFDFDFDFDFD5B00"
>       exs = run_bytecode(hexcode)

tests\test_cli.py:117: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src\halmos\__main__.py:424: in run_bytecode
    print(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <encodings.cp1252.IncrementalEncoder object at 0x000001A402AC6750>
input = 'Final opcode: STOP | Return data: None | Input example: \u2205'
final = False

    def encode(self, input, final=False):
>       return codecs.charmap_encode(input,self.errors,encoding_table)[0]
E       UnicodeEncodeError: 'charmap' codec can't encode character '\u2205' in position 56: character maps to <undefined>

C:\hostedtoolcache\windows\Python\3.11.4\x64\Lib\encodings\cp1252.py:19: UnicodeEncodeError
```

Reply:
got 'em with [edf439f](https://github.com/a16z/halmos/pull/124/commits/edf439f640d70961d0bf6139252d91b9cc9f4619)

<img width="812" alt="image" src="https://github.com/a16z/halmos/assets/85039585/b930723c-00d5-4b53-b7d3-5a8caa52046c">